pkg/model/keycloak_deployment.go

```diff
@@ -245,9 +245,39 @@ func KeycloakDeployment(cr *v1alpha1.Keycloak, dbSecret *v1.Secret) *v13.Statefu
 	} else if cr.Spec.MultiAvailablityZones.Enabled {
 		keycloakStatefulset.Spec.Template.Spec.Affinity = KeycloakPodAffinity(cr)
 	}
+
+	setSSLDBConnection(&keycloakStatefulset.Spec.Template.Spec, dbSecret)
+
 	return keycloakStatefulset
 }
 
+// In case SSLMODE property is not nil, we'll configure an env variable and a mount to the CRT secret
+// to connect to the DB using SSL
+func setSSLDBConnection(podSpec *v1.PodSpec, dbSecret *v1.Secret) {
+	if dbSecret != nil && dbSecret.Data[DatabaseSecretSslModeProperty] != nil {
+		podSpec.Containers[0].Env = append(podSpec.Containers[0].Env, v1.EnvVar{
+			Name:  DatabaseConnectionParamsProperty,
+			Value: "ssl=true;sslMode=" + string(dbSecret.Data[DatabaseSecretSslModeProperty]),
+		})
+
+		// We also expect that the secret with the CRT is present in the namespace
+		podSpec.Volumes = append(podSpec.Volumes, v1.Volume{
+			Name: DatabaseSecretSslCert + "_vol",
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName: DatabaseSecretSslCert,
+					Optional:   &[]bool{true}[0],
+				},
+			},
+		})
+		podSpec.Containers[0].VolumeMounts = append(podSpec.Containers[0].VolumeMounts, v1.VolumeMount{
+			Name:      DatabaseSecretSslCert + "_vol",
+			ReadOnly:  true,
+			MountPath: "/home/jboss/.postgresql",
```

Comment:
While this is valid for RH-SSO, please note that Keycloak image needs a different mount path: `/opt/jboss/.postgresql`

Reply:
:+1: